chalk-solve/src/recursive/fulfill.rs

```diff
@@ -421,8 +421,9 @@ impl<'s, 'db, I: Interner> Fulfill<'s, 'db, I> {
         // inference as an ambiguous solution.
 
         let interner = self.solver.program.interner();
+        let canonical_subst = self.infer.canonicalize(interner, &subst);
 
-        if self.infer.is_trivial_substitution(interner, &subst) {
+        if canonical_subst.quantified.value.is_identity_subst(interner) {
```

Comment:
This fixes #429, so can you add the test from that issue to `projection.rs`?

Reply:
done!